### ISSUE TITLE
Contact sync fails with "Not multi device, aborting..." while group sync succeeds.

### ISSUE THREAD

**Issue by liliakai** [signalapp/Signal-Android#6505]
MultiDeviceContactUpdateJob may fail silently with:
```
W/MultiDeviceContactUpdateJob(27054): Not multi device, aborting...
```
But the device *is* linked to Signal-Desktop. A MultiDeviceGroupUpdateJob running at the same time completes successfully.

See logs from WhisperSystems/Signal-Desktop#1107:

Debug log from desktop client (Chromium on Ubuntu 16.04): https://gist.github.com/c3f1e7ecc08cef3c733f3cf1e29d3b61

Debug log from HTC Thunderbolt, immediately after trying an import:
https://gist.github.com/605e4a43930a131e2f3a5622506d1243

**Comment by moxie0** [comment]
Where do you see it complete successfully? In the log it looks like a queued MultiDeviceGroupUpdateJob gets executed, which doesn't do a local check for link/unlink status, and it's told by the server that the device no longer exists. I don't see any indication that a desktop is actually linked here.

**Comment by liliakai** [comment]
The Desktop log shows that it consistently receives a [group sync](https://gist.github.com/anonymous/c3f1e7ecc08cef3c733f3cf1e29d3b61#file-debuglog-txt-L69) message but no contact sync message.

**Comment by moxie0** [comment]
The original issue says something about the "desktop app lost the link to the android app." I think there's something to investigate there: his local app state thinks that he never linked a desktop.

**Comment by liliakai** [comment]
Note that these logs are not from a later commenter, and not the OP. 

Where do you see the device no longer exists? I see a 410 (stale devices) but no 409 (missing/extra devices).

If the clients aren't linked, how was the desktop client able to send and trigger the multi device jobs at all?

**Comment by scottnonnenberg** [comment]
Here's another, more recent example of this bug (Android `4.12.3`, Desktop `1.0.37`): https://github.com/WhisperSystems/Signal-Desktop/issues/1208#issuecomment-343679631

**Comment by scottnonnenberg** [comment]
Seems to be the same as https://github.com/WhisperSystems/Signal-Android/issues/7175, which was fixed.

**Comment by scottnonnenberg** [comment]
We've got another user running into the 'Not multi device, aborting...' error: https://github.com/WhisperSystems/Signal-Desktop/issues/1927#issuecomment-355297921

Android log here: https://gist.github.com/anonymous/3a54dfef234fd4a27617782f8255e64c

**Comment by chadcatlett** [comment]
@scottnonnenberg I'm hitting the issue too on, but with a twist from what I can tell

Mobile: Android 8.1.0, Signal 4.14.10
Desktop: Debian sid, Signal Desktop 1.1.0

On Android I get the error about not being a multi device/linked, but interestingly the Desktop Client thinks it synced successfully.

If you'd like I can generate debugging logs and link them in this issue and the Desktop issue 1927.

**Comment by scottnonnenberg** [comment]
@chadcatlett Yes, logs would really help. When you say that Desktop thought it was successful, what scenario are you talking about? A new install or re-link, or manually requesting a contact sync in the settings dialog?

**Comment by chadcatlett** [comment]
@scottnonnenberg on an existing link that was previously working. If I manually click sync in the settings dialog the log shows it's successful in the desktop client, but the Android client's log has the message about not being a multi device.

As soon as I get a few minutes I'll create new logs and upload them as a gist.

**Comment by dualscyther** [comment]
Yep as mentioned, I'm running into this, but I just fixed it somehow:
https://github.com/WhisperSystems/Signal-Desktop/issues/1927#issuecomment-355297921

# History

1. Installed Signal Android on Nexus 5 stock ROM.
1. Installed Signal Desktop on Ubuntu 17.04, and synced contacts with no problems.
1. Installed Signal Android on Xiaomi Redmi 4X stock ROM (MIUI)
    - I didn't do any unlinking or uninstalling on my Nexus 5 before I performed this new install and signed in with my existing phone number.
    - At this point, I was still using Signal Desktop on Ubuntu 17.04, but I had to relink my phone with Signal Desktop. I was able to link, but unable to sync contacts (ran into this bug but didn't report it yet).
1. Switched from Ubuntu 17.04 to Debian Buster. Installed Signal Desktop and successfully linked with my phone, but still unable to sync contacts.
1. Just tried unlinking and relinking. It synced! Was this just a matter of "turning it off and back on"??? I'm gonna feel like an idiot if it was.
    - Desktop log: https://gist.github.com/b8ab67f2df8087b2372d51684cdd9915
    - Mobile log: https://gist.github.com/anonymous/3ca484cac27ad2a8d5fa298402b6c528

# Repost of my previous comment

---

Am also getting this. I tried to import contacts, it tried for a while and failed, then I submitted the two debug logs as soon as possible. Mention me for any other necessary details.

Android debug log:
https://gist.github.com/anonymous/3a54dfef234fd4a27617782f8255e64c

Debian Buster debug log:
https://gist.github.com/3b4e28403e36ee639f82afc3c298d56f

## Other info

- I have been messaging people on Signal Desktop regardless of no contact sync (it still shows their usernames and phone numbers)
- Night theme
- No passphrase
- No message trimming
- Installed Android app on Play store
- Using Xiaomi MIUI rom
 
---

**Comment by scottnonnenberg** [comment]
@dualscyther Seems that your Android install on your Xiaomi Redmi needed some encouragement to acknowledge that it was linked. All three of these scenarios were with the same Android install:
- (3) Android fresh install/Desktop re-link.
- (4) Android new link/Desktop fresh install - logs provided.
- The successful step was (5) was Android unlink then link/Desktop re-link - logs provided.

(4) is especially surprising. In that case it seems you were adding a new desktop instance, since you hadn't unlinked the old one. That seems like it would be the simplest, most straightforward scenario.

**Comment by SenorFusion** [comment]
@scottnonnenberg 

I was having this issue as well. Groups would sync but Contacts would not.

-Fresh install of Android 8.1.0 (January Security Patch), but is Custom Rom
-Fresh install of Debian 9

 I was able to solve it and I wanted to let you know how I fixed it. Unfortunately I tried 2 things at once and it worked, so I don't know which it was.

1. Removed Device, re-synced, during resync setup I gave the Device (my computer) a different name than I used the first time when it did not work. This was just a hunch.

2. My default Android Contacts app was being blocked from having network access via [AFWall+](https://github.com/ukanth/afwall). This has never been a problem before, including on Android 8.1.0 and Ubuntu 16.04.
